<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>579</width>
    <height>249</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SG384</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>IP:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="IP_address">
       <property name="text">
        <string>10.1.1.65</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Port:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="port">
       <property name="text">
        <string>5025</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_11">
       <property name="text">
        <string>Status:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="connection_status">
       <property name="text">
        <string>Disconnected</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="connect_button">
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QFrame" name="frame">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="0" rowspan="2">
       <widget class="QGroupBox" name="power_sweep_group">
        <property name="title">
         <string>Power settings</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="Vpp_spinbox">
           <property name="maximum">
            <double>20.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="mW_spinbox">
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>mW</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Vp-p</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>dBm</string>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QLineEdit" name="dBm_step_size">
           <property name="text">
            <string>0.1</string>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QLineEdit" name="mW_step_size">
           <property name="text">
            <string>1</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QLineEdit" name="Vpp_step_size">
           <property name="text">
            <string>1</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="dBm_spinbox">
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>30.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="2" colspan="2">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Step size</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QPushButton" name="read_power_button">
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="text">
            <string>Read
from device</string>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QCheckBox" name="auto_power_apply_checkbox">
           <property name="text">
            <string>Auto
apply</string>
           </property>
          </widget>
         </item>
         <item row="4" column="3">
          <widget class="QPushButton" name="apply_power_button">
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="text">
            <string>Apply
to device</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QGroupBox" name="power_sweep_group_2">
        <property name="title">
         <string>Frequency setting</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="2" colspan="3">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Step size</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QCheckBox" name="auto_freq_apply_checkbox">
           <property name="text">
            <string>Auto
apply</string>
           </property>
          </widget>
         </item>
         <item row="1" column="4">
          <widget class="QLineEdit" name="freq_step_size">
           <property name="text">
            <string>1</string>
           </property>
          </widget>
         </item>
         <item row="2" column="4">
          <widget class="QPushButton" name="apply_freq_button">
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="text">
            <string>Apply
to device</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QComboBox" name="freq_unit">
           <property name="currentIndex">
            <number>2</number>
           </property>
           <item>
            <property name="text">
             <string>Hz</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>KHz</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>MHz</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>GHz</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="freq_spinbox">
           <property name="decimals">
            <number>6</number>
           </property>
           <property name="maximum">
            <double>10000000000.000000000000000</double>
           </property>
           <property name="value">
            <double>1000.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="read_freq_button">
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="text">
            <string>Read
from device</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QFrame" name="frame_2">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>81</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Output status:</string>
         </property>
        </widget>
        <widget class="QLabel" name="output_status">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>10</y>
           <width>20</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="pixmap">
          <pixmap resource="../ui_resources/resources.qrc">:/icons/led-red-on.png</pixmap>
         </property>
         <property name="scaledContents">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QPushButton" name="output_button">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>10</y>
           <width>121</width>
           <height>23</height>
          </rect>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="text">
          <string>Turn on</string>
         </property>
        </widget>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="IDN_label">
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../ui_resources/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>Vpp_spinbox</sender>
   <signal>editingFinished()</signal>
   <receiver>Dialog</receiver>
   <slot>Vpp_editing_finished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Vpp_step_size</sender>
   <signal>editingFinished()</signal>
   <receiver>Dialog</receiver>
   <slot>Vpp_step_size_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mW_spinbox</sender>
   <signal>editingFinished()</signal>
   <receiver>Dialog</receiver>
   <slot>mW_editing_finished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mW_step_size</sender>
   <signal>editingFinished()</signal>
   <receiver>Dialog</receiver>
   <slot>mW_step_size_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dBm_spinbox</sender>
   <signal>editingFinished()</signal>
   <receiver>Dialog</receiver>
   <slot>dBm_editing_finished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>170</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dBm_step_size</sender>
   <signal>editingFinished()</signal>
   <receiver>Dialog</receiver>
   <slot>dBm_step_size_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>freq_spinbox</sender>
   <signal>editingFinished()</signal>
   <receiver>Dialog</receiver>
   <slot>freq_editing_finished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>288</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>freq_unit</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>Dialog</receiver>
   <slot>freq_unit_changed(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>462</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>freq_step_size</sender>
   <signal>editingFinished()</signal>
   <receiver>Dialog</receiver>
   <slot>freq_step_size_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>476</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>connect_button</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>connect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>568</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>output_button</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>output_on_off()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>558</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>read_power_button</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>read_power()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>66</x>
     <y>202</y>
    </hint>
    <hint type="destinationlabel">
     <x>277</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>apply_power_button</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>apply_power()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>202</y>
    </hint>
    <hint type="destinationlabel">
     <x>277</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>read_freq_button</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>read_freq()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>277</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>apply_freq_button</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>apply_freq()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>488</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>277</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Vpp_step_size_changed()</slot>
  <slot>Vpp_editing_finished()</slot>
  <slot>mW_editing_finished()</slot>
  <slot>mW_step_size_changed()</slot>
  <slot>dBm_editing_finished()</slot>
  <slot>dBm_step_size_changed()</slot>
  <slot>connect()</slot>
  <slot>output_on_off()</slot>
  <slot>freq_editing_finished()</slot>
  <slot>freq_step_size_changed()</slot>
  <slot>freq_unit_changed(QString)</slot>
  <slot>read_power()</slot>
  <slot>apply_power()</slot>
  <slot>read_freq()</slot>
  <slot>apply_freq()</slot>
 </slots>
</ui>
